File: src/main/scala/com/ajjpj/adiagram/ADiagramMain.scala
package com.ajjpj.adiagram

import javafx.stage.Stage
import com.ajjpj.adiagram.model.DiagramIO


/**
 * @author arno
 */
object ADiagramMain extends App {
  javafx.application.Application.launch(classOf[ADiagramMain])
}

class ADiagramMain extends javafx.application.Application {
  override def start(stage: Stage) {
    DiagramIO.init(stage)
  }
}

File: src/main/java/com/ajjpj/adiagram/ADiagramMain.java
package com.ajjpj.adiagram;

import javafx.application.Application;
import javafx.scene.Scene;
import javafx.scene.canvas.Canvas;
import javafx.scene.canvas.GraphicsContext;
import javafx.scene.control.Label;
import javafx.scene.layout.BorderPane;
import javafx.scene.layout.StackPane;
import javafx.scene.paint.Color;
import javafx.stage.Stage;

public class ADiagramMain extends Application {

    @Override
    public void start(Stage stage) {
//        String javaVersion = System.getProperty("java.version");
//        String javafxVersion = System.getProperty("javafx.version");
//        Label l = new Label("Hello, JavaFX " + javafxVersion + ", running on Java " + javaVersion + ".");
//        Scene scene = new Scene(new StackPane(l), 640, 480);

        BorderPane root = new BorderPane();
        Scene scene = new Scene(root, 1500, 900);

        final Canvas canvas = new Canvas(250,250);
        GraphicsContext gc = canvas.getGraphicsContext2D();

        gc.setFill(Color.BLUE);
        gc.fillRect(75,75,100,100);

        root.setCenter(canvas);

        stage.setScene(scene);
        stage.show();
    }

    public static void main(String[] args) {
        launch();
    }

}